### DISCUSSION TITLE
Weapons missing in H&M Arsenal

### DISCUSSION THREAD

**Discussion by XDSenDX** [Vdauphin/HeartsAndMinds#1548] [Question and answer] [answered]
After adding these 2 mods all the machine guns disappeared https://steamcommunity.com/sharedfiles/filedetails/?id=1713801249 https://steamcommunity.com/sharedfiles/filedetails/?id=1947836596.

Hope to find the answer. I also started a discussion inside the ace discord.
Cheers.

**Comment by Vdauphin** [comment, marked as answer]
Hello,

H&M filters weapons depending on the selected player role. For example, if you are not a machine gunner, you won't see those weapons. Additionally, it removes enemy weapons.
You can disable it in mission parameter under loadout section: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L428-L433

Cheers

**Reply by XDSenDX** [reply]
I'm not sure about enemy weapons tho. I still see them in arsenal.

Anyways thanks for your help.

Cheers.

**Comment by XDSenDX** [comment]
Can i add some kinds of launchers for the mg class. There's some tripod mounts which belong to launcher and it doesn't appear for the mg.

Cheers.

**Reply by Vdauphin** [reply]
Hello,
I don't see easy solution on this:
- You could switch to only allowed weapons (`Full`) and fill the https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L21-L48 but all player will have the same arsenal
- Or add to mg one of this trait `[["AssaultRifle"], ["RocketLauncher"], ["MissileLauncher", "128 + 512"]]` to this
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/arsenal/trait.sqf#L48 you may want to tweak the https://community.bistudio.com/wiki/CfgAmmo_Config_Reference#aiAmmoUsageFlags but I don't know if it will work
Cheers

**Reply by Vdauphin** [reply]
It is this https://community.bistudio.com/wiki/CfgAmmo_Config_Reference#aiAmmoUsageFlags

**Reply by XDSenDX** [reply]
Got the thing working now.

Cheers.

**Reply by XDSenDX** [reply]
Sorry for mentioning this again but can i just add a custom weapon, let's say an mp5, which isn't allowed in trait.sqf to a specific role, medic to be exact.

Cheers.

**Reply by Vdauphin** [reply]
You can check if player is a medic directly in the https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/arsenal/data.sqf#L67 and add the weapon you want for this specific role
(I think)

```sqf
if (player getUnitTrait "medic") then {
    [_box, ["myMP5classname"]] call ace_arsenal_fnc_addVirtualItems;
};
```
